genesis/engine/sensors/camera.py

```diff
@@ -0,0 +1,936 @@
+
+        # Store in cache (rgb_arr is already (H, W, 3) for env 0)
+        n_envs = self._manager._sim._B
+        if n_envs == 0:
+            # Single environment case
+            self._shared_metadata.image_cache[self._idx][0] = rgb_tensor
+        else:
+            # Multi-env case: only render env 0, replicate for others
+            for i in range(n_envs):
+                self._shared_metadata.image_cache[self._idx][i] = rgb_tensor
+
+    def add_light(
+        self,
+        color=(1.0, 1.0, 1.0),
+        intensity=1.0,
+        radius=0.5,
+        pos=(0.0, 0.0, 5.0),
+        revert_dir=False,
+        double_sided=False,
+        cutoff=180.0,
+    ):
+        """
+        Add a mesh light to the scene for all raytracer cameras.
+
+        Parameters
+        ----------
+        color : tuple[float, float, float]
+            Light color RGB.
+        intensity : float
+            Light intensity.
+        radius : float
+            Radius of the (spherical) light source.
+        pos : tuple[float, float, float]
+            Light position.
+        revert_dir : bool
+            Whether to revert normal direction for mesh light.
+        double_sided : bool
+            Whether the mesh light is double-sided.
+        cutoff : float
+            Light cutoff angle.
+        """
+        # Use the same mesh-light path as the high-level Scene API by creating
+        # a temporary sphere morph and letting the visualizer convert it.
+        scene = self._manager._sim.scene
+        if scene.is_built:
+            gs.logger.warning("RaytracerCameraSensor.add_light() should be called before scene.build(). Ignoring.")
+            return
+
+        morph = gs.morphs.Sphere(pos=pos, radius=radius)
+        scene.add_mesh_light(
+            morph=morph,
+            color=(*color, 1.0),
+            intensity=intensity,
+            revert_dir=revert_dir,
+            double_sided=double_sided,
+            cutoff=cutoff,
+        )
+
+    @gs.assert_built
+    def attach(self, rigid_link, offset_T):
+        """Attach the camera to a rigid link in the scene."""
+        self._attached_link = rigid_link
+        self._attached_offset_T = torch.as_tensor(offset_T, dtype=gs.tc_float, device=gs.device)
+        # Keep the underlying visualizer camera in sync so its own attach logic works.
+        self._camera_obj.attach(rigid_link, offset_T)
+        self._stale = True
+
+    @gs.assert_built
+    def detach(self):
+        """Detach the camera from the currently attached rigid link."""
+        self._attached_link = None
+        self._attached_offset_T = None
+        self._camera_obj.detach()
+        self._stale = True
+
+    @gs.assert_built
+    def move_to_attach(self):
+        """Move the camera to follow the currently attached rigid link."""
+        # Delegate to the underlying visualizer camera implementation
+        self._camera_obj.move_to_attach()
```

Comment:
Remove this.

Reply:
why? the backend is responsible for the attach method